src/wp-includes/media.php

```diff
@@ -5614,3 +5853,62 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * Add fetchpriority='high' to $loading_attrs if applicable.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param array[] $loading_attrs Array of the loading attributes for the element.
+ * @param array[] $attr          Array of the attributes for the element.
+ * @return array[] $loading_attrs Updated loading attributes for the element.
+ */
+function wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr ) {
+	if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority'] ) {
+		$loading_attrs['fetchpriority'] = 'high';
+		wp_high_priority_element_flag( false );
+		return $loading_attrs;
+	}
```

Comment:
The way you changed this is similarly problematic to the change for a custom-provided `loading` attribute. Please refer to my original suggestion in https://github.com/WordPress/wordpress-develop/pull/4495#discussion_r1213482656, we have to decouple two things here:
* Set the priority flag to false for images with custom `fetchpriority="high"`. You do that here already.
* Respect any `fetchpriority` value that is custom-provided and do not override it. Your code is currently not doing that: If someone provides `fetchpriority="low"`, our logic may still change it to `fetchpriority="high"`, which should not be the case.

So those have to be 2 _separate_ conditions.
```suggestion
	if ( isset( $attr['fetchpriority'] ) ) {
		if ( 'high' === $attr['fetchpriority'] ) {
			$loading_attrs['fetchpriority'] = 'high';
			wp_high_priority_element_flag( false );
		}
		return $loading_attrs;
	}
```

Reply:
This is addressed.